File: Iteration-2/src/NonTechnicalElective.java
import java.util.ArrayList;
import org.apache.log4j.Logger;

public class NonTechnicalElective extends ElectiveCourse {
        private Logger log = Logger.getLogger(NonTechnicalElective.class);

        public NonTechnicalElective(String courseName, String courseCode, int courseCredit, int courseDay,
                        String courseHour, int quato, ArrayList<Integer> semesters) {
                super(courseName, courseCode, courseCredit, courseDay, courseHour, quato, semesters);
                log.info(this.getCourseName() + " (" + this.getCourseCode() + ")"
                                + " named nontechnical elective course created.");
        }

        @Override
        public boolean isEligibleToRequest(Student student) {
                /* log.info(student.getStudentName() + " can "
                                + ((semesterControl(student))
                                                ? ""
                                                : "not")
                                + "enroll this course: "
                                + this.getCourseName()); */
                return semesterControl(student);
        }
}

File: Iteration-3/Code/NonTechnicalElective.py
from ElectiveCourse import ElectiveCourse
from typing import List

class NonTechnicalElective(ElectiveCourse):
    def __init__(self, courseName: str, courseCode: str, courseCredit: int, courseDay: int,
                 courseHour: str, quota: int, semesters: List[int]):
        super().__init__(courseName, courseCode, courseCredit, courseDay, courseHour, quota, semesters)

    def isEligibleToRequest(self, student) -> bool:
        return self.semesterControl(student)
